flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImpl.java

```diff
@@ -60,89 +63,78 @@
 public class ChannelStateWriterImpl implements ChannelStateWriter {
 
     private static final Logger LOG = LoggerFactory.getLogger(ChannelStateWriterImpl.class);
-    private static final int DEFAULT_MAX_CHECKPOINTS =
-            1000; // includes max-concurrent-checkpoints + checkpoints to be aborted (scheduled via
-    // mailbox)
```

Comment:
Hi @pnowojski 

From the comment, we can get that the `ConcurrentMap<Long, ChannelStateWriteResult> results;` saves ongoingCheckpoints and abortedCheckpoints. 

But the checkpoints that were aborted before start were not saved. We want aborted checkpoints to ignore start. So I introduced the ` NavigableSet<Long> abortedCheckpointIds;` to save all aborted checkpointIds. 

- If checkpointId is aborted before start, it will be saved into this set. 
- When the start is called, if checkpointId in this set, the start should be ignored.

And introduced the `long ongoingCheckpointId;` to save the ongoing checkpointId. Because the concurrent unaligned checkpoint isn't supported, so the `long` is enough. And checkpoints with checkpointId less than `ongoingCheckpointId` should be aborted. So I remove these checkpointIds when the ongoingCheckpointId is updated. 

What do you think?

Reply:
Hi @pnowojski , thanks for your review.

The `checkpointId < nextExpectedCheckpointId` may happen. For example, `AlternatingWaitingForFirstBarrierUnaligned#barrierReceived` don't check whether checkpointId is aborted, if received a barrier, it will call the ChannelStateWriter.start().

So, we should use `ongoingCheckpointId`, and just update it within `ChannelStateWriterImpl#start`

We should ignore checkpoint when the checkpointId is aborted, so we  need the `NavigableSet<Long> abortedCheckpointIds;`.

Reply:
Thanks for the explanation. But in that case, why do we need `abortedCheckpointIds` set? Couldn't we have just `long ongoingCheckpointId` and `long nextExpectedCheckpointId`?

Reply:
Hi @pnowojski , great idea! 

Updated, but I used maxAbortedCheckpointId instead of nextExpectedCheckpointId. It should be clearer.